src/services/search.ts

```diff
@@ -0,0 +1,29 @@
+// parent search. holds logic for splitting between different services
+
+import * as play from 'play-dl';
+import { PlayValidation } from '../types/definitions';
+
+// TODO: handle search for other sources. for now, default to yt_videos only
+export const search = async (
+    query: string,
+): Promise<play.YouTubeVideo | null> => {
+    const validationResult: PlayValidation = await play.validate(query);
+
+    if (validationResult === 'search') {
```

Comment:
Might be better to use a switch statement.
Also, does validate return a string or an enum?
Maybe prefer using the enum if it exists.

Reply:
https://github.com/shinray/vythm/blob/3aa4f30a68aca60efa9debe59802e376d486f476/src/types/definitions.d.ts#L23

I have it as another union type here. thoughts?

Reply:
basically it's because the library also returns it as a union type
it includes a `false` as well, not just a bunch of strings...